XML/XMLExtension.cs

```diff
@@ -124,21 +124,72 @@ public static void CreateXMLFile(List<InfixExpression> expressions)
         }
 
         /// <summary>
-        /// This method opens the xml file to be viewed on browser
+        /// This method asks what browser the user would like to use.
         /// </summary>
         public static void OpenXMLInBrowser()
         {
+
+            bool done = false;
+            string browser = "";
+            Console.WriteLine("The browser choices that you have to view the XML file are:" +
+                "\n\t1) Chrome" +
+                "\n\t2) Firefox" +
+                "\n\t3) Edge" +
+                "\nOr type Q to quit...");
+            do
+            {
+                Console.Write("\nWhat browser would you like to view the XML file in?: ");
+                string selection = Console.ReadLine().ToLower();
+                switch (selection)
+                {
+                    case ("1"):
+                        done = ProcessStarted("chrome", "Chrome");
+                        break;
+                    case ("2"):
+                        done = ProcessStarted("firefox", "FireFox");
+                        break;
+                    case ("3"):
+                        done = ProcessStarted("msedge", "Microsoft Edge");
+                        break;
+                    case ("q"):
+                        done = true;
+                        break;
+                    default:
+                        Console.WriteLine("\nInvalid selection. ");
+                        break;
+                }
+            } while (!done);
+
+
+        }
+
+        /// <summary>
+        /// Attempt to open the xml file in the user's selected browser
+        /// </summary>
+        /// <param name="browser">the command line argument to open the selected browser.</param>
+        /// <param name="browserName">The browser name.</param>
+        /// <returns>The result.</returns>
+        public static bool ProcessStarted(string cmdBrowser, string browserName)
+        {
+            bool started = false;
             try
             {
-                string path = Path.Combine(Environment.CurrentDirectory, FILE_PATH);
+
+                string path = cmdBrowser == "firefox"? FILE_PATH : Path.Combine(Environment.CurrentDirectory,FILE_PATH);
```

Comment:
Why is firefox different?

Reply:
just tested this out, they don't need to be different if you use backslashes in the FILE_PATH instead of forward. So switch FILE_PATH to be `const string FILE_PATH = @"..\..\..\Data\Project_2_Info_5101_XML_Output.xml";` instead and that works

Reply:
I just did a quick update of this, should be working fine now